1.1  Extract Information From Your Own Transaction

In [3]:
import requests
import datetime

#extracting the block height from my handcash transactions
block_height = '866234'  

# collecting block data from Whatsonchain API
api_url = f"https://api.whatsonchain.com/v1/bsv/main/block/height/{block_height}"
response = requests.get(api_url)

#checking the status of the data
if response.status_code == 200:
    block_data = response.json()
    print("Block Data (JSON Format):")
    print(block_data)
    
    # Obtaining the required information
    tx_count = block_data.get("txcount", "N/A")
    unix_time = block_data.get("time", "N/A")
    total_fee = block_data.get("totalFees", "N/A")
    confirmation = block_data.get("confirmations", "N/A")
    miner = block_data.get("miner", "N/A")
    
    # Convert Unix timestamp to human readable format
    human_time = datetime.datetime.fromtimestamp(unix_time).strftime('%Y-%m-%d %H:%M:%S') if unix_time != "N/A" else "N/A"

    print("\nExtracted Information:")
    print(f"Transaction Count (txcount): {tx_count}")
    print(f"Time (Human-Readable): {human_time}")
    print(f"Total Fees: {total_fee}")
    print(f"Confirmations: {confirmation}")
    print(f"Miner: {miner}")
else:
    print(f"Error fetching block data: {response.status_code}")


Block Data (JSON Format):
{'hash': '00000000000000000aebb32793cb81c9449d109c9d3783590d13d31b2e465703', 'confirmations': 13094, 'size': 958735, 'height': 866234, 'version': 771751936, 'versionHex': '2e000000', 'merkleroot': '4728d569f8251f6135ff5603d16bc5be8ecead6d9dadc46817343a970061b540', 'txcount': 1794, 'nTx': 0, 'num_tx': 1794, 'tx': ['409802088faaa67e8a504ece2458e93d77179b9a443e76d5576f21cb54e4f9ec', '485e26bea603ee6d0b8f80526ece16467d1c6fff4d8720821a81473671b1a90e', '28a6e3a575a1fc373eb77ed48d85da904a61668cc0071871be677cb27692a066', 'f33764d6dbd128b999c43a9534db1acdf12c2fe597227594fc7c3f972fdd65c2', 'c0a217046d693d55e3c82980b171df9762fe86651e40a94607d3fcff60187256', '2ae7e5dd8552fe38ad222646e42d1ed1c5232b36e838624b167a9ec8ed5484ed', 'b82b87d9b74e05faf7e6c7e507eada95b2d8256aa714736f9c63d063557a4713', '77767951bc2cf02b886706a75f9ebcf5b1172f3219318d518aebfc43eebc69c1', '55b6fdb5df18e133f7c91f8632f965034cc4e176ca32186fbddf313bfb58d383', 'ce5eb635a7f95185e8f2503ea973e6368fa70071f33fbe

Transaction count (txcount):
(Value = 1794)
This represents the number of transactions included within the block. Transactions are particular records of cryptocurrency transfer between addresses. A high transaction count shows that this block is busy,containing many transfers.

Time (Human-Readable):
(Value: 2024-10-13 18:57:14)
This is the timestamp when the block was mined, stored as a Unix timestamp. Additionally, it shows the conversion of the Unix timestamp-seconds since January 1, 1970-to a human-readable date/time in UTC. It helps track when this block was created.

Total Fees:
(Value: 0.0003310199999999597 BTC)
This is the sum of all transaction fees in the block. Miners collect these fees as a reward for processing and confirming transactions within the block. The value reflects the sum of all the fees contributed by every transaction in the block.

Confirmations:
(Value: 12791)
This is the amount of blocks mined after the block in subject. A block with more confirmations is more confident about the security of its transactions.The block data becomes more immutably with every confirmation of a block.

Miner:
(Value: taal.com)
This is the entity, or mining pool, that completed the mining of a block. The competition among miners involves cryptography-based puzzles about adding a block to the blockchain, the solution of which has a prize in the block reward and transaction fees.